specs/core/0_beacon-chain.md

```diff
@@ -342,14 +346,23 @@ class HistoricalBatch(Container):
     state_roots: Vector[Root, SLOTS_PER_HISTORICAL_ROOT]
 ```
 
+#### `DepositMessage`
+
+```python
+class DepositMessage(Container):
+    pubkey: BLSPubkey
+    withdrawal_credentials: Bytes32
+    amount: Gwei
+```
+
 #### `DepositData`
 
 ```python
 class DepositData(Container):
     pubkey: BLSPubkey
     withdrawal_credentials: Bytes32
     amount: Gwei
-    signature: BLSSignature
+    signature: BLSSignature  # signing over DepositMessage
```

Comment:
Can you clarify this comment? What does `signing over DepositMessage` mean?

Reply:
`DepositData` is a special case:
- The signature is included as part of the data in the deposit-datas hash-tree-root. And we can not completely separate it from the message as with the other containers, since that would not be compatible with the existing verified deposit contract code. The contract expects `H(H(p,w), H(a,s))`, not `H(H(H(p,w),H(a,0)), s)` (first characters of field names).
- The signature is not included when creating the message hash to sign (because the signature does not exist yet at that point). That is what `DepositMessage` is for: to replicate the old signing-root behavior of hash-tree-root over all fields except the signature.
Also see `Deposit`(merkle proof + signature + contents)/`DepositData`(signature + contents)/`DepositMessage`(contents) naming discussion above in this PR. 

It's not pretty, but deposits are not broken this way, and we can have all the benefits as outlined in #1487 to make up for it.